### PR TITLE
handle all our new strings correctly, introduce rotary_pronunciation

### PR DESCRIPTION
**Pull request by MoKob**
# Issue

https://github.com/Project-OSRM/osrm-backend/issues/2869

## Tasklist
 - [x] add regression / cucumber cases
 - [x] review
 - [x] adjust for for comments

### CODE REVIEW

File: src/engine/api/json_factory.cpp [hunk 1]
```diff
@@ -240,7 +240,13 @@ util::json::Object makeRouteStep(guidance::RouteStep step, util::json::Value geo
     if (!step.destinations.empty())
         route_step.values["destinations"] = std::move(step.destinations);
     if (!step.rotary_name.empty())
+    {
         route_step.values["rotary_name"] = std::move(step.rotary_name);
+        if( !step.rotary_pronunciation.empty() )
```
Comment (daniel-j-h) [inline src/engine/api/json_factory.cpp]: clang-format

**Comment by daniel-j-h** [comment]
`docs/http.md` missing for rotary name / pronunciation.

**Comment by daniel-j-h** [comment]
@bsudekum @1ec5 to keep you in the loop here: rotaries can have additional fields in the response:

- `rotary_name`
- `rotary_pronunciation`

**Comment by MoKob** [comment]
From chat, documentation isn't missing. Merging

**Pull request merged**